### DISCUSSION TITLE
Filter embedded resources filter only the nexted. I expected to filter the first level

### DISCUSSION THREAD

**Discussion by calebeaires** [supabase/supabase#2207] [Questions] [answered]
My intention is to filter values from a primary table using a nested field as the pivot of the filter. Let me explain with this example:

A. This is the select with the filter.
```js
const test = async () => {
  let {
    data,
    err
  } = await supabases
    .from('messages')
    .select('id, user_id, users(id, username)')
    .filter('users.username', 'eq', 'Calebe')
    console.log(data, err)
}

test()

```

B. The result gives by Supabase is this one:

```js
[{
  id: 1,
  user_id: "8d0fd2b3-9ca7-4d9e-a95f-9e13dded323e",
  users: {
    id: "8d0fd2b3-9ca7-4d9e-a95f-9e13dded323e",
    username: "Calebe"
  }
}, {
  id: 2,
  user_id: "8d0fd2b3-9ca7-4d9e-a95f-9e13dded323a",
  users: null
}]
```

C. This is what I expected to get (👍🏻    🎉)

```js
[{
  id: 1,
  user_id: "8d0fd2b3-9ca7-4d9e-a95f-9e13dded323e",
  users: {
    id: "8d0fd2b3-9ca7-4d9e-a95f-9e13dded323e",
    username: "Calebe"
  }
}]
```

The second item comes by default when using supabase simple filter. But I expected more atomic results considering the filter will exclude the result at the primary level.

**Comment by steve-chavez** [comment, marked as answer]
Hey @calebeaires,

This is a pending enhancement being tracked at: https://github.com/supabase/postgrest-js/issues/197. A workaround for now, would be to filter the rows based on `users: null` on the client side.

**Reply by calebeaires** [reply]
Yup. This is a good solution. I will follow that issue since I can't use this approach on the table that could lead to big dataset.

**Reply by cvincentcoleman** [reply]
This isn't working for me.

**Comment by spencercap** [comment]
@calebeaires 
not sure how new this is, but what you want to do is now available without filtering all the results client-side like so:

```js
const { data, error } = await supabase
  .from('users')
  .select()
  .eq('address->postcode', 90210)
```

https://supabase.com/docs/reference/javascript/using-filters#filter-by-values-within-a-json-column

**Reply by cvincentcoleman** [reply]
Could you show this in a way that relates to this example? I'm not seeing that arrow represented anywhere in the docs
